receiver/snmpreceiver/scraper.go

```diff
@@ -39,13 +64,345 @@ func newScraper(logger *zap.Logger, cfg *Config, settings component.ReceiverCrea
+func getScalarDataPointAttributes(configHelper *configHelper, oid string) map[string]string {
+	dataPointAttributes := map[string]string{}
+	for _, attribute := range configHelper.metricAttributesByOID[oid] {
+		attributeKey := attribute.Name
+		if value := configHelper.getAttributeConfigValue(attributeKey); value != "" {
+			attributeKey = value
+		}
+		dataPointAttributes[attributeKey] = attribute.Value
+	}
+
+	return dataPointAttributes
+}
+
+// getIndexedDataPointAttributes gets attributes for this metric's datapoint based on the previously
+// gathered attributes.
+// Keys will be determined from the related attribute config and values will come a few
+// different places.
+// Enum attribute value - comes from the metric config's attribute data
+// Indexed prefix attribute value - comes from the current SNMP data's index and the attribute
+// config's prefix value
+// Indexed OID attribute value - comes from the previously collected indexed attribute data
+// using the current index and attribute config to access the correct value
+func getIndexedDataPointAttributes(
+	configHelper *configHelper,
+	parentOID string,
+	indexString string,
+	indexedAttributeValues map[indexedAttributeKey]string,
+) (map[string]string, error) {
+	datapointAttributes := map[string]string{}
+
+	for _, attribute := range configHelper.getMetricConfigAttributes(parentOID) {
+		attributeName := attribute.Name
+
+		var attributeValue string
+		prefix := configHelper.getAttributeConfigIndexedValuePrefix(attributeName)
+		oid := configHelper.getAttributeConfigOID(attributeName)
+		switch {
+		case prefix != "":
+			attributeValue = prefix + indexString
+		case oid != "":
+			indexedAttrKey := indexedAttributeKey{
+				parentOID: oid,
+				oidIndex:  indexString,
+			}
+			attributeValue = indexedAttributeValues[indexedAttrKey]
+		default:
+			attributeValue = attribute.Value
+		}
+
+		// If no good attribute value could be found
+		if attributeValue == "" {
+			return nil, errors.New(errMsgAttributeEmptyValue)
+		}
+
+		attributeKey := attributeName
+		if value := configHelper.getAttributeConfigValue(attributeKey); value != "" {
+			attributeKey = value
+		}
+		datapointAttributes[attributeKey] = attributeValue
+	}
+
+	return datapointAttributes, nil
+}
+
+// getResourceAttributes creates a map of key/values for all related resource attributes. Keys
+// will come directly from the metric config's resource attribute values. Values will come
+// from the related attribute config's prefix value plus the index OR the previously collected
+// resource attribute indexed data.
+func getResourceAttributes(
+	configHelper *configHelper,
+	parentOID string,
+	indexString string,
+	indexedResourceAttributeValues map[indexedAttributeKey]string,
+) (map[string]string, error) {
+	resourceAttributes := map[string]string{}
+
+	for _, attributeName := range configHelper.getResourceAttributeNames(parentOID) {
+		var attributeValue string
```

Comment:
This variable ends up being an unnecessary allocation. Would be easy enough to assign it directly to the map.

Reply:
@djaglowski because not making changes due to the above comment, this is still needed

Reply:
@djaglowski again i'll make this change, but i don't really see this as necessary for the code to make it into the repo.

Reply:
> @djaglowski again i'll make this change, but i don't really see this as necessary for the code to make it into the repo.

This is an unproductive sentiment that you have expressed many times throughout the review of this component. The readability and maintainability of the code in this component has improved dramatically since your initial PR. While you may not fully appreciate this, I think in the future you could save a lot of time for yourself and whoever is sponsoring your contribution by placing more importance on these aspects of code quality.